apps/bookmark/js/bookmark_editor.js

```diff
@@ -70,6 +73,25 @@ var BookmarkEditor = {
     var icon = new Icon(this.bookmarkIcon, this.data.icon);
     icon.render({'size': this.BOOKMARK_ICON_SIZE});
   },
+
+  _renderAppIcon: function renderAppIcon(manifest, manifestURL, size) {
+    // Parse icon URL from app manifest
+    var iconURL = window.WebManifestHelper.iconURLForSize(manifest,
+      manifestURL, size);
+    if (!iconURL) {
+      return;
+    }
+    // Switch out placeholder for real icon when it loads
+    this.appIconListener = this._handleAppIconLoad.bind(this);
+    this.appIcon.addEventListener('load', this.appIconListener);
+    this.appIcon.setAttribute('src', iconURL);
```

Comment:
OK, I took a look at the code in /shared/js/homescreens/icon.js

As part of bug 1134657 we're going to want to change that code to use App.getIcon in Gecko to retrieve icons for installed apps rather than do an XHR from Gaia. That won't work for this patch because it's not an installed app so my conclusion is that it's probably better for this use to be implemented separately until such a time as we have a new API to let Gecko safely fetch the icon for not-yet-installed apps. I hope that's OK.